dask/array/tests/test_linalg.py

```diff
@@ -842,19 +843,51 @@ def test_no_chunks_svd():
 
 @pytest.mark.parametrize("shape", [(10, 20), (10, 10), (20, 10)])
 @pytest.mark.parametrize("chunks", [(-1, -1), (10, -1), (-1, 10)])
-def test_svd_flip(shape, chunks):
+@pytest.mark.parametrize("dtype", ["f4", "f8"])
+def test_svd_flip_correction(shape, chunks, dtype):
     # Verify that sign-corrected SVD results can still
     # be used to reconstruct inputs
-    x = da.random.random(size=shape, chunks=chunks)
+    x = da.random.random(size=shape, chunks=chunks).astype(dtype)
     u, s, v = da.linalg.svd(x)
 
+    # Choose precision in evaluation based on float precision
+    decimal = 9 if np.dtype(dtype).itemsize > 4 else 6
+
     # Validate w/ dask inputs
     uf, vf = svd_flip(u, v)
-    np.testing.assert_almost_equal(np.asarray(np.dot(uf * s, vf)), x, decimal=9)
+    assert uf.dtype == u.dtype
+    assert vf.dtype == v.dtype
+    np.testing.assert_almost_equal(np.asarray(np.dot(uf * s, vf)), x, decimal=decimal)
 
     # Validate w/ numpy inputs
     uc, vc = svd_flip(*da.compute(u, v))
-    np.testing.assert_almost_equal(np.asarray(np.dot(uc * s, vc)), x, decimal=9)
+    assert uc.dtype == u.dtype
+    assert vc.dtype == v.dtype
+    np.testing.assert_almost_equal(np.asarray(np.dot(uc * s, vc)), x, decimal=decimal)
+
+
+@pytest.mark.parametrize("dtype", ["f2", "f4", "f8", "f16", "c8", "c16", "c32"])
+@pytest.mark.parametrize("u_based", [True, False])
+def test_svd_flip_sign(dtype, u_based):
+    if sys.platform == "win32" and dtype in ["f16", "c32"]:
+        pytest.skip("128-bit floats not supported on windows")
+    # fmt: off
```

Comment:
I think we've avoided `fmt`  pragmas so far. I have a slight preference to just accept black's formatting here, even if it's less readable in this case.

Reply:
Sounds good: https://github.com/dask/dask/pull/6658/commits/e73d9f8740454665a17ccfde2bd998c3b8d882fc